XMonad/Layout/CenteredIfSingle.hs

```diff
@@ -36,30 +36,36 @@ import XMonad.Layout.LayoutModifier
 --
 -- and adding the 'centeredIfSingle' layoutmodifier to your layouts.
 --
--- > myLayoutHook = centeredIfSingle 0.7 Grid ||| ...  
+-- > myLayoutHook = centeredIfSingle 0.7 0.8 Grid ||| ...
 --
 -- For more information on configuring your layouts see "XMonad.Doc.Extending".
 
 
--- | Layout Modifier that places a window in the center of the screen, 
+-- | Layout Modifier that places a window in the center of the screen,
 -- leaving room on the left and right if there is only a single window
-newtype CenteredIfSingle a = CenteredIfSingle Double deriving (Show, Read)
+data CenteredIfSingle a = CenteredIfSingle Double Double deriving (Show, Read)
 
 instance LayoutModifier CenteredIfSingle Window where
-  pureModifier (CenteredIfSingle ratio) r _ [(onlyWindow, _)] = ([(onlyWindow, rectangleCenterPiece ratio r)], Nothing)
+  pureModifier (CenteredIfSingle ratioX ratioY) r _ [(onlyWindow, _)] = ([(onlyWindow, rectangleCenterPiece ratioX ratioY r)], Nothing)
   pureModifier _ _ _ winRects = (winRects, Nothing)
 
--- | Layout Modifier that places a window in the center of the screen, 
--- leaving room on the left and right if there is only a single window
+-- | Layout Modifier that places a window in the center of the screen,
+-- leaving room on all sides if there is only a single window
 centeredIfSingle :: Double -- ^ Ratio of the screen the centered window should take up. Should be a value between 0.0 and 1.0
+                 -> Double -- ^ Vertical ratio
                  -> l a    -- ^ The layout that will be used if more than one window is open
                  -> ModifiedLayout CenteredIfSingle l a
-centeredIfSingle ratio = ModifiedLayout (CenteredIfSingle ratio)
+centeredIfSingle ratioX ratioY = ModifiedLayout (CenteredIfSingle ratioX ratioY)
 
 -- | Calculate the center piece of a rectangle given the percentage of the outer rectangle it should occupy.
-rectangleCenterPiece :: Double -> Rectangle -> Rectangle
-rectangleCenterPiece ratio (Rectangle rx ry rw rh) = Rectangle start ry width rh
+rectangleCenterPiece :: Double -> Double -> Rectangle -> Rectangle
+rectangleCenterPiece ratioX ratioY (Rectangle rx ry rw rh) = Rectangle startX startY width height
   where
-    sides = floor $ fi rw * (1.0 - ratio) / 2
-    start = fi rx + sides
-    width = fi $ fi rw - sides * 2
+    left = floor $ fi rw * (1.0 - ratioX) / 2
+    top = floor $ fi rh * (1.0 - ratioY) / 2
+
+    startX = fi rx + left
+    width = fi $ fi rw - left * 2
+
+    startY = fi ry + top
+    height = fi $ fi rh - top * 2
```

Comment:
There's a lot of repetition here, maybe we should take care of that

Reply:
@TheMC47 I made some changes here: [e576f85](https://github.com/xmonad/xmonad-contrib/pull/725/commits/e576f85bf32820b328c768a0228ac3c3167bf5ec)

Reply:
Awesome, thanks